### DISCUSSION TITLE
What does this do: `programs.nixvim.path`?

### DISCUSSION THREAD

**Discussion by refaelsh** [nix-community/nixvim#2066] [Q&A] [answered]
Hi,

Regarding this entry in the docs: https://nix-community.github.io/nixvim/NeovimOptions/index.html#path.
What does it do? It is not explained.

Thanks in advance.

**Comment by MattSturgeon** [comment, marked as answer]
It is used for [`files`](https://nix-community.github.io/nixvim/NeovimOptions/index.html#files) sub-modules to define where the generated file will be installed. E.g. you can create ftplugins using nixvim modules:

```nix
programs.nixvim = {
  enable = true;

  files.foo = {
    path = "ftplugin/python.lua";
    opts.expandtab = true;
  };
};
```

However in submodules it defaults to `name`, so usually you just define the attr name:


```nix
programs.nixvim = {
  enable = true;

  files."ftplugin/python.lua" = {
    opts.expandtab = true;
  };
};
```

I can't recall OTTOMH if it is used outside of `files` submodules, if not we should probably mark it hidden or internal. Either way we could improve its docs.
